roadrunner.rb

```diff
@@ -0,0 +1,131 @@
+
+      # RoadRunner currently doesn't support kilobyte units. Converting to megabytes instead.
+      if result['unit'] == 'Kilobytes'
+        result['unit'] = 'Megabytes'
+        if result['measurements'].is_a?(Array)
+          result['measurements'].each_with_index do |measurement, index|
+            result['measurements'][index] = measurement / 1000.0
+          end
+        else
+          result['measurements'] = result['measurements'] / 1000.0
+        end
+      end
+      results << result
+    end
+    results
+  end
+
+  def self.generate_result_name(benchmark)
+    split = benchmark.split('_')
+    split.delete('mem')
+    split.delete('time')
+    split.delete('size')
+    split[-1] = 'time' if split[-1] == 'ms'
+    split[-1] = 'size' if split[-1] == 'kb'
+    case split[0]
+    when 'gem', 'require', 'client'
+      split.join('.')
+    else
+      idx = split.index('small') || split.index('large')
+      if idx
+        "#{split[0...idx].join}.#{split[(idx + 1)..-1].join('.')}"
+      elsif split.include?('time')
+        "#{split[0...-1].join}.#{split[-1]}"
+      elsif split.include?('size')
+        "#{split[0...-2].join}.#{split[-2]}.#{split[-1]}"
+      else
+        split.join
+      end
+    end
+  end
+
+  def self.generate_description(benchmark)
+    split = benchmark.split('_')
+    split.delete('mem')
+    split.delete('time')
+    split.delete('size')
+    split[-1] = 'time' if split[-1] == 'ms'
+    split[-1] = 'size' if split[-1] == 'kb'
+    case split[0]
+    when 'gem'
+      'The size of the gem.'
+    when 'require'
+      if split.include?('time')
+        'The time it takes to require the gem.'
+      else
+        "The amount of memory #{split[1]} when requiring the gem."
+      end
+    when 'client'
+      if split.include?('time')
+        'The time it takes to initialize the client.'
+      else
+        "The amount of memory #{split[1]} when creating the client."
+      end
+    else
+      idx = split.index('small') || split.index('large')
+      if idx
+        operation = split[0...idx].map(&:capitalize).join
+        if split.include?('time')
+          "The time it takes to perform the #{operation} operation."
+        else
+          "The amount of memory allocated to perform the #{operation} operation."
+        end
+      elsif split.include?('time')
+        operation = split[0...-1].map(&:capitalize).join
+        "The time it takes to perform the #{operation} operation."
+      elsif split.include?('size')
+        operation = split[0...-2].map(&:capitalize).join
+        "The amount of memory allocated to perform the #{operation} operation."
+      else
+        split.join(' ')
+      end
+    end
+  end
+
+  def self.unit(benchmark)
+    if benchmark.include?('kb')
+      'Kilobytes'
+    elsif benchmark.include?('ms')
+      'Milliseconds'
+    end
+  end
+
+  def self.generate_dimensions(key, shared_dimensions)
+    dimensions = shared_dimensions.dup
```

Comment:
The base dimensions should probably be defined here. I don't see this changing anywhere else?